queue-manager/rango-preset/src/helpers.ts

```diff
@@ -1456,10 +1460,22 @@ export function retryOn(
     finalQueueToBeRun = onlyWalletMatched[0];
   }
 
-  if (!network || !canSwitchNetworkTo?.(wallet, network)) {
-    finalQueueToBeRun?.unblock();
-  } else {
-    finalQueueToBeRun?.checkBlock();
+  if (finalQueueToBeRun) {
+    const finalQueueStorage = finalQueueToBeRun.getStorage() as SwapStorage;
+    const currentSwap = getCurrentStep(finalQueueStorage?.swapDetails);
+
+    const currentNamespace = currentSwap
+      ? getCurrentNamespaceOfOrNull(finalQueueStorage.swapDetails, currentSwap)
+      : null;
+    if (
+      !network ||
+      !currentNamespace ||
+      !canSwitchNetworkTo?.(wallet, network, currentNamespace)
+    ) {
+      finalQueueToBeRun?.unblock();
```

Comment:
```suggestion
      finalQueueToBeRun.unblock();
```
No need for optional chaining anymore